### DISCUSSION TITLE
How to use local library snapshot?

### DISCUSSION THREAD

**Discussion by dpproduction** [cashapp/paparazzi#1453] [Q&A] [answered]
I would like to use library from special branch, but I don't understand which files I should put to project...
I have compiled library, so there are many JARs like paparazzi-1.3.4-SNAPSHOT.jar, paparazzi-gradle-plugin-1.3.4-SNAPSHOT.jar, paparazzi-preview-processor-1.3.4-SNAPSHOT.jar. 
Help me please which files needed for use Paparazzi locally ?

**Comment by TWiStErRob** [comment, marked as answer]
As far as I know: all of them.

I think your best bet is to:
 * do a `gradlew publishToMavenLocal`
 * adjust your repository listing to add `mavenLocal()`,
 * so that you can use it as "normal",
 * by just bumping the version number where necessary.

---

This is **almost** how the tests work in the repo:

https://github.com/cashapp/paparazzi/blob/a2e9b03eaf281ff336bd7dd307acb4f092933007/build.gradle#L70-L75

https://github.com/cashapp/paparazzi/blob/a2e9b03eaf281ff336bd7dd307acb4f092933007/paparazzi-gradle-plugin/build.gradle#L51

https://github.com/cashapp/paparazzi/blob/a2e9b03eaf281ff336bd7dd307acb4f092933007/paparazzi-gradle-plugin/src/test/projects/test.settings.gradle#L8-L11

but Gradle tests are **special**, normally you can achieve this with just `mavenLocal()`. Use the above quoted code only if you're not fine with the default `~/.m2` location.

**Reply by dpproduction** [reply]
Thanks for the answer. 
I have tried this way but faced with a problem: 
I call publishToMavenLocal then copy repo folder to my location.
Part of my module screenshot-testing build script:

```
buildscript {
    repositories {
        mavenCentral()
        google()
        maven {
            url "file://${rootDir}/screenshot-testing/localMaven"
        }
    }
    dependencies {
        classpath 'app.cash.paparazzi:paparazzi-gradle-plugin:1.3.5-SNAPSHOT'
    }
}
```
![Снимок экрана от 2024-06-04 16-32-20](https://github.com/cashapp/paparazzi/assets/5700728/762c3099-7c21-4fc0-91bb-aa9d14365328)
![Снимок экрана от 2024-06-04 16-32-11](https://github.com/cashapp/paparazzi/assets/5700728/09354756-c6cd-484f-8ff5-6151b4b57f88)

Gradle sync writes - Failed to resolve: app.cash.paparazzi:paparazzi:1.3.5-SNAPSHOT
Seems like it tries to load library from remote but not from local repo...

**Reply by TWiStErRob** [reply]
It seems that the plugin was resolved, which makes sense because you have `buildscript` repository, add the same repository for standard dependency resolution too. Notice that the code block you show only has 3 repos, while the error shows 6-8.
